src/libcore/mem/maybe_uninit.rs

```diff
@@ -509,32 +509,189 @@ impl<T> MaybeUninit<T> {
         self.as_ptr().read()
     }
 
-    /// Gets a reference to the contained value.
+    /// Gets a shared reference to the contained value.
+    ///
+    /// This can be useful when we want to access a `MaybeUninit` that has been
+    /// initialized but don't have ownership of the `MaybeUninit` (preventing the use
+    /// of `.assume_init()`).
     ///
     /// # Safety
     ///
-    /// It is up to the caller to guarantee that the `MaybeUninit<T>` really is in an initialized
-    /// state. Calling this when the content is not yet fully initialized causes undefined
-    /// behavior.
+    /// Calling this when the content is not yet fully initialized causes undefined
+    /// behavior: it is up to the caller to guarantee that the `MaybeUninit<T>` really
+    /// is in an initialized state.
+    ///
+    /// # Examples
+    ///
+    /// ### Correct usage of this method:
+    ///
+    /// ```rust
+    /// #![feature(maybe_uninit_ref)]
+    /// use ::std::mem::MaybeUninit;
+    ///
+    /// let mut x = MaybeUninit::<Vec<u32>>::uninit();
+    /// // Initialize `x`:
+    /// unsafe { x.as_mut_ptr().write(vec![1, 2, 3]); }
+    /// /* The above line can also be done without unsafe:
+    /// x = MaybeUninit::new(vec![1, 2, 3]); // */
```

Comment:
The `// */` here looks odd, what is that about?

Reply:
I just wanted to show a line of Rust that compiled ignoring the "comment" context of the above line. But now with a fresh look it does look strange 😅 
Anyways, I don't think we will be keeping both options in _this_ documentation exampl (maybe in the main `struct` documentation?). So, for this example, which one do you think is best? The one without `unsafe` is tempting, but often people will use `MaybeUninit` with out-pointer initialization patterns, so the one with the `.write()` seems more realistic (and is generally the example shown in the documentation of other methods).

Reply:
Yeah, I think I prefer the `write`.